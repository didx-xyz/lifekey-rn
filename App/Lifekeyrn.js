/**
 * Lifekey App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import * as Lifecycle from './Lifecycle'
import Logger from './Logger'
import Palette from './Palette'
import Api from './Api'
import Session from './Session'
import Routes from './Routes'
import Config from './Config'
import ConsentUser from './Models/ConsentUser'
import ConsentConnection from './Models/ConsentConnection'
import ConsentConnectionRequest from './Models/ConsentConnectionRequest'
import ConsentDiscoveredUser from './Models/ConsentDiscoveredUser'
import ConsentISA from './Models/ConsentISA'
import EventEmitter from 'EventEmitter'
import React, { Component } from 'react'
import {
  View,
  Dimensions,
  Navigator,
  DeviceEventEmitter,
  Platform,
  StatusBar
} from 'react-native'

const PORTRAIT = 0
const LANDSCAPE = 1

export default class Lifekeyrn extends Component {

  constructor(props) {
    super(props)
    // Members
    this._className = this.constructor.name
    this.filename = this._className + '.js'
    this._navigationEventEmitter = new EventEmitter()
    this._orientationEventEmitter = new EventEmitter()
    this.state = {
      booted: false,
      orientation: null,
      viewableScreenWidth: null,
      viewableScreenHeight: null,
      screenWidth: null,
      screenHeight: null,
      scale: null,
      fontScale: null
    }
    Logger.info(` === ${Config.APP_NAME}  v${Config.version} === `, this.filename)

    // Events
    if (Platform.OS === 'android') {
      DeviceEventEmitter.addListener('messageReceived', (message) => this._nativeEventMessageReceived(message))
      DeviceEventEmitter.addListener('tokenRefreshed', (token) => this._nativeEventTokenRefreshed(token))
      // lol

    } else {
      Logger.info('TODO: Firebase iOS', this.filename)
    }
    this._initSession()
    this._initialRoute = this._getInitialRoute()

  }

  _getInitialRoute() {
    if (Config.initialRouteFromConfig) {
      return Config.initialRoute
    } else {
      const userState = Session.getState().user

      if (!userState || !userState.registered) {
        return Routes.onboarding.splashScreen
      } else {
        return Routes.onboarding.unlock
      }
    }
  }

  _initSession() {
    Logger.info('Initialising session', this.filename)
    // We want to essentially look through storage
    // to determine the starting point of global state
    ConsentUser.get()
    .then(results => {
      if (results) {
        const update = {}
        update[ConsentUser.storageKey] = {
          id: results.id,
          did: results.did,
          registered: results.registered || false,
          loggedIn: results.loggedIn || false,
          email: results.email,
          firebaseToken: results.firebaseToken
        }
        this.setState({
          booted: true
        }, () => {
          Session.update(update)
          this.forceUpdate()
        })

      } else {
        const update = {}
        update[ConsentUser.storageKey] = {
          registered: false,
          loggedIn: false,
        }

        this.setState({
          booted: true
        }, () => {
          Session.update(update)
          console.log('BOOTED TRUE')
          this.forceUpdate()

        })

      }
    })
    .catch(error => {
      Logger.error('Error restoring session', this.filename, error)
    })
  }

  /**
   * Fires when a Firebase EventMessage is received
   * @param {string} message The firebase message
   */
  _nativeEventMessageReceived(message) {
    Logger.info('Firebase message received', this.filename)
    console.log(JSON.stringify(message))

    if (message.data && message.data.type) {
      switch (message.data.type) {

      case 'received_did':
        Logger.firebase('received_did')
        ConsentUser.setDid(message.data.did_value)
        .then(did => {
          Logger.info(`DID set to ${did}`, this.filename)
        })
        .catch(error => {
          Logger.error('Could not set DID', this.filename, error)
        })
        break

      case 'user_connection_request':
        Logger.firebase('user_connection_request')
        ConsentConnectionRequest.add(
          message.data.user_connection_request_id,
          message.data.from_id,
          message.data.from_did,
          message.data.from_nickname
        )
        .then(connectionRequests => {
          Logger.firebase('ConsentConnectionRequest updated')
          console.log(connectionRequests)
        })
        .catch(error => {
          Logger.error('Error writing to ConsentConnectionRequest', this.filename, error)
        })
        ConsentDiscoveredUser.add(
          message.data.from_id,
          message.data.from_did,
          message.data.from_nickname
        )
        .then(discoveredUsers => {
          Logger.firebase("ConsentDiscoveredUser updated")
          console.log(discoveredUsers)
        })
        .catch(error => {
          Logger.error('Error writing to ConsentDiscoveredUser ')
        })
        break

      case 'user_connection_created':
        Logger.firebase('user_connection_created')
        ConsentConnection.add(
          message.data.user_connection_id,
          message.data.to_id
        )
        .then(() => {
          Logger.info('Connection created')
          this.forceUpdate()
          Logger.info('Force update')
        })
        .catch(error => {
          Logger.error(error, this.filename, error)
        })
        break
      case 'sent_activiation_email':
        Logger.firebase('sent_activiation_email')
        alert(`${message.notification.title} - ${message.notification.body}`)
        break
      case 'app_activation_link_clicked':
        Logger.firebase('app_activation_link_clicked')
        this.navigator.replace(Routes.main)
        break
      case 'information_sharing_agreement_request':
        Logger.firebase('information_sharing_agreement_request')
        ConsentISA.add(
          message.data.isar_id,
          message.data.from_id
        )
        .then(() => {
          Logger.info('ISA Added')
        })
        .catch(error => {
          Logger.error(error, this.filename, error)
        })
        break
      default:
        Logger.firebase(JSON.stringify(message))
        if (message.notification) {
          Logger.info(message.notification.title + ' - ' + message.notification.body, this.filename)
        }
        break
      }

    } else {
      // Just a normal notification
      if (message.notification) {
        Logger.firebase(JSON.stringify(message))
      } else {
        Logger.error('Unexpected firebase message format')
      }
    }
  }

  /**
   * Fires when firebase pushes a new token to the device
   * @param {string} token The firebase token
   * @returns {undefined} undefined
   */
  _nativeEventTokenRefreshed(token) {
    ConsentUser.setToken(token)
    .catch(error => {
      Logger.firebase(error)
    })
  }

  componentWillMount() {
    Logger.react(this.filename, Lifecycle.COMPONENT_WILL_MOUNT)

    // Get some initial size data
    const width = Math.round(Dimensions.get('window').width)
    const height = Math.round(Dimensions.get('window').height)
    const scale = Dimensions.get('window').scale
    const fontScale = Dimensions.get('window').fontScale
    Logger.info(`Screen Dimensions: ${width} x ${height} ${width < height ? 'PORTRAIT' : 'LANDSCAPE'}`, this.filename)

    // Set to state
    this.setState({
      screenWidth: width,
      screenHeight: height,
      orientation: width > height ? LANDSCAPE : PORTRAIT,
      scale: scale,
      fontScale: fontScale
    })
  }

  onWillFocus(route) {
    this._navigationEventEmitter.emit('onWillFocus' + route.scene.name)
  }

  onDidFocus(route) {
    this._navigationEventEmitter.emit('onDidfocus' + route.scene.name)
  }

  componentDidMount() {
    Logger.react(this.filename, Lifecycle.COMPONENT_DID_MOUNT)
  }

  componentWillReceiveProps() {
    Logger.react(this.filename, Lifecycle.COMPONENT_WILL_RECEIEVE_PROPS)
  }

  componentWillUpdate(nextProps, nextState) {
    Logger.react(this.filename, Lifecycle.COMPONENT_WILL_UPDATE)
  }

  componentDidUpdate() {
    Logger.react(this.filename, Lifecycle.COMPONENT_DID_UPDATE)
  }

  componentWillUnmount() {
    Logger.react(this.filename, Lifecycle.COMPONENT_WILL_UNMOUNT)

    // Remove event listeners
    DeviceEventEmitter.removeListener('messageReceived', (e) => this._nativeEventMessageReceived(e))
    DeviceEventEmitter.removeListener('tokenRefreshed', (token) => this._nativeEventTokenRefreshed(token))
  }

  shouldComponentUpdate(nextProps, nextState) {
    // Re-render when orientation changes
    if (nextState.viewableScreenWidth !== this.state.viewableScreenWidth ||
        nextState.viewableScreenHeight !== this.state.viewableScreenHeight) {
      return true
    }
    return false
  }

  /**
   * Called when the outermost view's dimensions change
   * @param {Object} event
   * @returns {undefined} undefined
   */
  _onScreenUpdate(event) {

    const width = Math.round(event.nativeEvent.layout.width)
    const height = Math.round(event.nativeEvent.layout.height)
    const orientation = width > height ? 'LANDSCAPE' : 'PORTRAIT'
    if (orientation !== this.state.orientation) {
      // emit orientation change event
      this._orientationEventEmitter.emit('orientation')
    }
    if (this.state.viewableScreenHeight !== height ||
        this.state.viewableScreenWidth !== width) {
      this.setState({
        viewableScreenWidth: width,
        viewableScreenHeight: height,
        orientation: orientation
      })
      Logger.info(`Usable screen: ${width} x ${height} ${orientation}`, this.filename)
    }
  }


  render() {
    return (
        <Navigator
          ref={(_navigator) => this.navigator = _navigator}
          initialRoute={this._initialRoute}
          onWillFocus={(route) => this.onWillFocus(route)}
          onDidFocus={(route) => this.onDidFocus(route)}
          renderScene={ (route, navigator) => {
            Logger.routeStack(navigator.getCurrentRoutes())
            return (
              <View
                onLayout={(event) => this._onScreenUpdate(event)}
                style={{ flex: 1, backgroundColor: Palette.sceneBackgroundColour }}
              >
                <StatusBar hidden={true} />
                {React.createElement(
                  route.scene,
                  {
                    booted: this.state.booted,
                    screenWidth: this.state.viewableScreenWidth,
                    screenHeight: this.state.viewableScreenHeight,
                    screenOrientation: this.state.screenOrientation,
                    scale: this.state.scale,
                    fontScale: this.state.fontScale,
                    route,
                    navigator,
                    _navigationEventEmitter: this._navigationEventEmitter, // Navigator events
                    passProps: route.passProps || {} // allow views to pass data to new views
                  }
                )}
              </View>
            )
          }}
          configureScene={ () =>
            ({
              ...Config.sceneConfig,
              gestures: {} // Prevents the user from being able to swipe to go back
            })
          }
        />
    );
  }
}
