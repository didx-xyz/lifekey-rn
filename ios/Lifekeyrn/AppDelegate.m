/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>
#import <Firebase.h>

#import "RNFirebaseMessaging.h"

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  [FIRApp configure];
  [[UNUserNotificationCenter currentNotificationCenter] setDelegate:self];
  
  NSURL *jsCodeLocation;

  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];

  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"Lifekeyrn"
                                               initialProperties:nil
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  
  return YES;
}

// add before @end
- (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions))completionHandler
{
//  [RNFirebaseMessaging willPresentNotification:notification withCompletionHandler:completionHandler];
}

- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler
{
//  [RNFirebaseMessaging didReceiveNotificationResponse:response withCompletionHandler:completionHandler];
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(nonnull NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler{
//  [RNFirebaseMessaging didReceiveRemoteNotification:userInfo fetchCompletionHandler:completionHandler];
}

@end
