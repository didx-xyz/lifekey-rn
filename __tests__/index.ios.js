import 'react-native'
import React from 'react'
import Lifekeyrn from '../App/Lifekeyrn'

jest.mock('react-native-camera')

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Lifekeyrn />
  )
  
  expect(tree).toMatchSnapshot()
})
