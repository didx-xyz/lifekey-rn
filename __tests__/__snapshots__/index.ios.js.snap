// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
              Object {
                "rotate": "0deg",
              },
              Object {
                "skewX": "0deg",
              },
              Object {
                "skewY": "0deg",
              },
            ],
          },
          null,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentContainerStyle={
              Object {
                "padding": 0,
              }
            }
            contentInset={
              Object {
                "bottom": 49,
              }
            }
            enableAutoAutomaticScroll={true}
            extraHeight={75}
            keyboardDismissMode="interactive"
            onScroll={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={8}
            showsVerticalScrollIndicator={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 28,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Developer Menu
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {},
                  ]
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#94de45",
                      "height": 50,
                      "justifyContent": "center",
                      "margin": 5,
                      "opacity": 1,
                      "padding": 10,
                      "paddingHorizontal": 20,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#fff",
                            "fontWeight": "600",
                            "paddingHorizontal": 8,
                          },
                          Object {
                            "fontSize": 12,
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    >
                      KEYSTORE MANAGER
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {},
                  ]
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#94de45",
                      "height": 50,
                      "justifyContent": "center",
                      "margin": 5,
                      "opacity": 1,
                      "padding": 10,
                      "paddingHorizontal": 20,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#fff",
                            "fontWeight": "600",
                            "paddingHorizontal": 8,
                          },
                          Object {
                            "fontSize": 12,
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    >
                      SELF-FACING CAMERA
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {},
                  ]
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#94de45",
                      "height": 50,
                      "justifyContent": "center",
                      "margin": 5,
                      "opacity": 1,
                      "padding": 10,
                      "paddingHorizontal": 20,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#fff",
                            "fontWeight": "600",
                            "paddingHorizontal": 8,
                          },
                          Object {
                            "fontSize": 12,
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    >
                      REGISTER ON CONSENT
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#fff",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 20,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Session
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  {}
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 20,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Storages
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 15,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Connections
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 15,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Connection Requests
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 15,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Discovered Users
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc1cc",
                        "fontSize": 15,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "paddingVertical": 10,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  User
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      Object {},
                    ]
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityLabel={undefined}
                    accessibilityTraits="button"
                    accessible={true}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#2f8cff",
                        "height": 50,
                        "justifyContent": "center",
                        "margin": 5,
                        "opacity": 1,
                        "padding": 10,
                        "paddingHorizontal": 20,
                      }
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#fff",
                              "fontWeight": "600",
                              "paddingHorizontal": 8,
                            },
                            Object {
                              "fontSize": 12,
                            },
                            undefined,
                            Object {},
                          ]
                        }
                      >
                        TEST CONNECT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
